### PR TITLE
New employee management improvement #18434

### PR DESCRIPTION
**Pull request by atm-steve**
Add read and write rights for user as employee, add two new fields ref_employee and national register number in RH and Bank tab, move Accounting account field to RH and Bank tab.

### CODE REVIEW

File: htdocs/install/mysql/tables/llx_user.sql [hunk 1]
```diff
@@ -108,5 +108,7 @@ create table llx_user
   import_key        varchar(14),				-- import key
   default_range     integer,
   default_c_exp_tax_cat     integer,
+  employee_number   varchar(50),
```
Comment (atm-maxime) [inline htdocs/install/mysql/tables/llx_user.sql:111]: Should be ref_employee

File: htdocs/user/bank.php [hunk 2]
```diff
@@ -502,6 +502,25 @@
...
 	}
 
+	// Accountancy code
+	if (!empty($conf->accounting->enabled)) {
+		print '<tr><td>'.$langs->trans("AccountancyCode").'</td>';
+		print '<td>'.$object->accountancy_code.'</td></tr>';
+	}
+
+	// Employee Number
+	if (!empty($conf->accounting->enabled)) {
```
Comment (atm-maxime) [inline htdocs/user/bank.php:512]: No need to test for accounting enabled

Comment (hregis) [reply htdocs/user/bank.php:512]: Hello @atm-maxime 

to see more broadly, if we accept to save the accounting data while the module is not active,
must we save the data of the other modules while they are not activated?
Should we keep a history or make sure to have the means to recreate this history?

File: htdocs/core/modules/modHRM.class.php [hunk 3]
```diff
@@ -249,6 +249,22 @@ public function __construct($db)
 		$this->rights[$r][4] = 'compare_advance';
 		$this->rights[$r][5] = 'read'; // In php code, permission will be checked by test if ($user->rights->hrm->compare_advance->read)
 		$r++;
+
+		// Read employee
+		$this->rights[$r][0] = 4031; // Permission id (must not be already used)
+		$this->rights[$r][1] = 'Read employee'; // Permission label
```
Comment (eldy) [inline htdocs/core/modules/modHRM.class.php:255]: An employee is a user with object. 
And we already have a permission to read users and a permission for each type of object (leave, expense report, salary, etc...).
Adding such a permission create a lot of business rule conflicts into the code and on some security checks.
i think the permission you want to add is not to define if we can see an employee or not but to decide if we can see some kind of information (like personal information) ?

Comment (atm-steve) [reply htdocs/core/modules/modHRM.class.php:255]: These two new rights is about the personal informations fields only.

Comment (eldy) [reply htdocs/core/modules/modHRM.class.php:255]: So can you please rename the label of permission so things are clear. They were not for me and i have to go into the code to understand the permission was to view/read personal information. But end users are not able to do this analyze so having a better label of permission will avoid interrogation.

**Comment by atm-maxime** [comment]
🥳

**Pull request merged**